SalesforceSDK/Core/Settings/IApplicationInformationService.cs

```diff
@@ -33,7 +33,7 @@ public interface IApplicationInformationService
     {
         Task<string> GetApplicationDisplayNameAsync();
 
-        Task<string> GenerateUserAgentHeaderAsync(bool isHybrid);
+        Task<string> GenerateUserAgentHeaderAsync(bool isHybrid, bool isSmartSync);
```

Comment:
Also the lowest level lib (SalesforceSDK) should not need to know about higher level libs (hybrid/smartsync).